imap_processing/ultra/l1b/de.py

```diff
@@ -144,47 +151,49 @@ def calculate_de(de_dataset: xr.Dataset, name: str) -> xr.Dataset:
         {key: de_dataset[dataset_key] for key, dataset_key in zip(keys, dataset_keys)}
     )
 
-    vx_ultra, vy_ultra, vz_ultra = get_unit_vector(
+    vx_ultra, vy_ultra, vz_ultra = get_de_velocity(
         (de_dict["x_front"], de_dict["y_front"]),
         (de_dict["x_back"], de_dict["y_back"]),
         de_dict["front_back_distance"],
         de_dict["tof_start_stop"],
     )
+    de_dict["direct_event_velocity"] = np.column_stack(
+        (
+            vx_ultra.astype(np.float32),
+            vy_ultra.astype(np.float32),
+            vz_ultra.astype(np.float32),
+        )
+    )
 
-    de_dict["vx_ultra"] = vx_ultra.astype(np.float32)
-    de_dict["vy_ultra"] = vy_ultra.astype(np.float32)
-    de_dict["vz_ultra"] = vz_ultra.astype(np.float32)
+    de_dict["tof_energy"] = get_de_energy_kev(
+        (vx_ultra, vy_ultra, vz_ultra), species_bin
+    )
+    de_dict["azimuth"], de_dict["elevation"] = get_de_az_el(
+        (vx_ultra, vy_ultra, vz_ultra)
+    )
     de_dict["energy"] = energy
     de_dict["species"] = species_bin
 
     # Annotated Events.
-    position = np.stack(
-        (de_dict["vx_ultra"], de_dict["vy_ultra"], de_dict["vz_ultra"]), axis=-1
-    )
-
     ultra_frame = getattr(SpiceFrame, f"IMAP_ULTRA_{sensor}")
     sc_velocity, sc_dps_velocity, helio_velocity = get_annotated_particle_velocity(
-        de_dataset.data_vars["EVENTTIMES"],
-        position,
+        de_dataset.data_vars["EVENTTIMES"].values,
+        de_dict["direct_event_velocity"],
         ultra_frame,
         SpiceFrame.IMAP_DPS,
         SpiceFrame.IMAP_SPACECRAFT,
     )
 
-    de_dict["vx_sc"], de_dict["vy_sc"], de_dict["vz_sc"] = (
-        sc_velocity[:, 0],
-        sc_velocity[:, 1],
-        sc_velocity[:, 2],
+    de_dict["velocity_sc"] = np.column_stack(
+        (sc_velocity[:, 0], sc_velocity[:, 1], sc_velocity[:, 2])
```

Comment:
I don't think that this column_stack call is doing anything.
```suggestion
    de_dict["velocity_sc"] = sc_velocity
```

Reply:
This is probably me just making sure python works for me. N00bish care.